testament/important_packages.nim

```diff
@@ -1,134 +1,136 @@
-template pkg1(name: string; hasDeps = false; cmd = "nimble test"; url = "", useHead = true): untyped =
-  packages1.add((name, cmd, hasDeps, url, useHead))
+template pkg1(name: string; cmd = "nimble test"; url = "", useHead = true): untyped =
+  packages1.add((name, cmd, url, useHead))
 
-template pkg2(name: string; hasDeps = false; cmd = "nimble test"; url = "", useHead = true): untyped =
-  packages2.add((name, cmd, hasDeps, url, useHead))
+template pkg2(name: string; cmd = "nimble test"; url = "", useHead = true): untyped =
+  packages2.add((name, cmd, url, useHead))
 
-var packages1*: seq[tuple[name, cmd: string; hasDeps: bool; url: string, useHead: bool]] = @[]
-var packages2*: seq[tuple[name, cmd: string; hasDeps: bool; url: string, useHead: bool]] = @[]
+var packages1*: seq[tuple[name, cmd: string; url: string, useHead: bool]] = @[]
+var packages2*: seq[tuple[name, cmd: string; url: string, useHead: bool]] = @[]
 
+# If url == "" useHead indicates wether nimble will install the latest head or release version
+# If url != "" useHead indicates wether nimble will be involved at all (useHead = false: nimble won't be involved)
```

Comment:
that sounds weird, why can't we do:

`nimble develop "https://github.com/jackmott/easygl"`

Reply:
Because nimble doesn't support that.

Reply:
how so? I just tried it and it seems to work

Reply:
Oh, nice, I tried with "https://krux02@bitbucket.org/krux02/tensordslnim.git" and that didn't work, so I assumed it didn't work at all, but I just had to remove the "krux02@" prefix; will adapt accordingly.
Btw @timotheecour can you look at the CI failures, its really weird

Reply:
Oh, I think I found the culprit :D

Reply:
```
/Users/runner/work/Nim/Nim/testament/testament.nim(825) testament
  /Users/runner/work/Nim/Nim/testament/testament.nim(781) main
  /Users/runner/work/Nim/Nim/testament/categories.nim(715) processCategory
  /Users/runner/work/Nim/Nim/testament/categories.nim(517) testNimblePackages
  /Users/runner/work/Nim/Nim/testament/testament.nim(110) execCmdEx2
  /Users/runner/work/Nim/Nim/lib/pure/osproc.nim(990) startProcess
  /Users/runner/work/Nim/Nim/lib/pure/osproc.nim(1058) startProcessAuxSpawn
  /Users/runner/work/Nim/Nim/lib/pure/os.nim(1390) setCurrentDir
  /Users/runner/work/Nim/Nim/lib/pure/includes/oserr.nim(94) raiseOSError
  Error: unhandled exception: No such file or directory
  Additional info: "pkgstemp/nake" [OSError]
```

let (installCmdLine, installOutput, installStatus) = execCmdEx2("nimble", ["develop", if useHead: name & "@#head" else: name, "-y"])
would call
`nimble develop nake -y`

this creates nake and pkgs (not sure exactly why, that might've changed from before where IIRC it was only installing package, not creating pkgs; not 100% sure)


then 
`let (buildCmdLine, buildOutput, buildStatus) = execCmdEx2(cmdArgs[0], cmdArgs[1..^1], workingDir=buildPath)`

would fail since `workingDir=buildPath` doesn't exist (since let packagesDir = "pkgstemp")

which should explain why CI fails

Reply:
Hmm, it doesn't seem to be that fleshed out yet (it clones into "jackmott/easygl" for example, not into "easygl" directly, and in general the ability to test packages that lack a .nimble file seems useful)

Reply:
Oh, and then theres the issue that you can't develop binary only packages :/

Reply:
And another issue, nimble develop will always use head, so that won't work either. (And never did)

Reply:
ok; so how about:
```
export NIMBLE_DIR=/tmp/nim_ci_nimbledir # avoid messing up user's local machine
# for each pkg:
git clone url destdir
cd destdir
if a binary package:
  nimble build
else:
  nimble install -y --depsOnly
run test cmd (nimble test or the one in important_packages)
```

Reply:
Using git clone now, (useHead = false will use the latest tag) and nimble install --depsOnly